code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -464,3 +434,335 @@
 
 	M.adjustToxLoss(5 * REM * delta_time)
 	M.adjustOrganLoss(ORGAN_SLOT_LIVER, 3 * REM * delta_time)
+
+/datum/reagent/drug/mushroomhallucinogen
+	name = "Mushroom Hallucinogen"
+	description = "A strong hallucinogenic drug derived from certain species of mushroom."
+	color = "#E700E7" // rgb: 231, 0, 231
+	metabolization_rate = 0.2 * REAGENTS_METABOLISM
+	taste_description = "mushroom"
+	ph = 11
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	addiction_types = list(/datum/addiction/hallucinogens = 12)
+
+/datum/reagent/drug/mushroomhallucinogen/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	if(!M.slurring)
+		M.slurring = 1 * REM * delta_time
+	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "tripping", /datum/mood_event/high, name)
+	switch(current_cycle)
+		if(1 to 5)
+			if(DT_PROB(5, delta_time))
+				M.emote(pick("twitch","giggle"))
+		if(5 to 10)
+			M.Jitter(10 * REM * delta_time)
+			if(DT_PROB(10, delta_time))
+				M.emote(pick("twitch","giggle"))
+		if (10 to INFINITY)
+			M.Jitter(20 * REM * delta_time)
+			if(DT_PROB(16, delta_time))
+				M.emote(pick("twitch","giggle"))
+	..()
+
+/datum/reagent/drug/mushroomhallucinogen/on_mob_metabolize(mob/living/L)
```

Comment:
please name `L` better

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names